processing/src/main/java/io/druid/query/search/search/SearchQuery.java

```diff
@@ -254,6 +275,10 @@ public boolean equals(Object o)
       return false;
     }
 
+    if (!strategy.equals(that.strategy)) {
```

Comment:
Hash code is already done here. 
https://github.com/druid-io/druid/pull/3792/files/f8d5c5ee8cb9d536017f6aaa538dbdc21f2ff9d3#diff-0cd94720b9f3bbe8caa4e047dc3bb5e5R295